d1/david-rev1.py

```diff
@@ -0,0 +1,23 @@
+"""
+d1 solution using stack/queue object
+"""
+from collections import deque
+
+
+def solve(mylist):
+    tmplist = []
+    count = len(mylist)/2
+    stack = mylist
+    queue = deque(mylist)
```

Comment:
[Disclaimer: I don't really know python so take this as grain of salt.]
I think `deque` is what you need, and I just look up `deque` in python, and it's beautifully done.
I would say you don't actually need tmplist (which IMO would violate the "must be in place" requirement)

When you write
`queue = deque(mylist)`

queue becomes a deque object. Different from Python’s general purpose built-in containers, `dict`, `list`, `set`, and `tuple`,`deque` is a specialized container datatypes. Therefore, it get some special methods that are very useful for this problem.

The `popleft` is a good one, and you could do `pop` for getting the right side of `deque`. That way if the first arg of function is `stack`, which is the given iterable, you can use that with `queue` to compute the solution without allocating memory for other variables. 

Hope that makes sense. Thanks for submitting in python, push me to learn and look up documentation.

Reply:
great feedback! new submission coming incorporating feedback.

def v3(queue):
    stack = []
    while len(queue) > 0:
        try:
            stack.append(queue.popleft())
            stack.append(queue.pop())
        except IndexError:
            break
    return stack


if __name__ == "__main__":
    queue = deque([1,2,3,4,5])
    print(v3(queue))